### PR TITLE
[Go] Client Models using AllOf, OneOf, or AnyOf Correctly Include `time` and `os` imports

### PR DESCRIPTION
**Pull request by icubbon**
Addresses https://github.com/OpenAPITools/openapi-generator/issues/13959

Currently, if a Model is an `allOf` the `time` and `os` imports are not correctly added to the generated file. This was introduced recently with a fix to not include those imports when the model is a `composedSchema` https://github.com/OpenAPITools/openapi-generator/pull/13833. The logic in that fix was just slightly off as an `allOf` should be treated the same as a standard model.

If a model is an `AllOf` or does not have any composed schemas at all, the sub-models are in-lined defined in the struct. In this case, the standard logic of including the `time` and `os` imports apply.

If a model is a `OneOf` or `AnyOf`, the sub-models are included as pointers to the defined model. In this case, do not include those items in the logic of including `time` and `os` imports.

Added an example to samples/openapi3/client/petstore/go/go-petstore/api/openapi.yaml that covers the missing case.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@antihax @grokify @kemokemo @jirikuncar @ph4r5h4d @wing328

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,1214 +1,322 @@
-<h1 align="center">OpenAPI Generator</h1>
```
Comment (wing328) [inline README.md:1]: @icubbon thanks for the PR. Why deleted the project's README ?

Comment (icubbon) [reply README.md]: @wing328 The `ci/circleci: node1` tests are failing with the below message. Is this a flaky test framework setup issue?
```
Execution default of goal org.springframework.boot:spring-boot-maven-plugin:3.0.0:repackage failed: Unable to load the mojo 'repackage' in the plugin 'org.springframework.boot:spring-boot-maven-plugin:3.0.0' due to an API incompatibility: org.codehaus.plexus.component.repository.exception.ComponentLookupException: org/springframework/boot/maven/RepackageMojo has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

**Comment by viswajithiii** [comment]
Hello! We are running into this issue as well. Will this PR be merged soon? 🙂 Let me know if I can help with anything.

**Comment by icubbon** [comment]
Gonna try a new PR without the updated example to see if that will pass CI

**Comment by vitaly-zdanevich** [comment]
Why closed? I have a problem that `time` is not imported.

**Pull request closed without merging**